src/extension.ts

```diff
@@ -621,9 +630,9 @@ export async function activate(context: vscode.ExtensionContext) {
                                         break;
                                 }
                             } catch (err) {
-                                console.log(
-                                    `Non-JSON output from the process :  ${message}`
-                                );
+                                // if not a JSON string, it is incomplete
```

Comment:
Is there a way to handle this without catching an error (as part of the normal flow), but instead verifying if this caracter is present "\0"

Reply:
At this point in the process, the main string from the python side is already split by "\0", so it wouldn't be present. 

To be clear, only individual JSON strings are added to the incompletion buffer. this means that, if the message was 

`{ test: one } \0 { test: two`

`{test:one}` and `{test:two` would be processed separately. Only `{test:two` would be added to the incompletion buffer.

Reply:
Ok, I realize I was looking at the catch for the wrong error. It is not optimal to have the catch as part of the normal flow, but after quick look there is no easy way to detect if the string is a valid json or not

Reply:
Yes, it's pretty tricky to do it without looping it having the code jumping everywhere.